### ISSUE TITLE
combine all rows into single row in mozilla, chrome is working fine

### ISSUE THREAD

**Issue by muhammad-Ali-Nawaz** [react-csv/react-csv#167]

**Comment by mccabemj** [comment]
Can you provide more Information regarding this? The library doesn’t do any row combining so not sure how it could be the libraries issue

**Comment by muhammad-Ali-Nawaz** [comment]
const csvData = data.length ? data.map(user=>{
                return {
                    'First Name': user.firstName,
                    'Last Name': user.lastName,
                    'Phone': user.mobile,
                    'Email': user.email,
                    'Registration Date': user.createdAt,
                    'Sparks': user.pointsInWallet,
                    'Deals': user.wishlistCount,
                    'Redemptions': user.redeemedDeals,
                }
            }) : [{'First Name':'','Last Name':'','Phone':'','Email':'','Registration Date':'','Sparks':'','Deals': '','Redemptions':'' }];
            this.setState({csvData}, ()=>{
                this.csvRef.link.click();
            });

**Comment by mccabemj** [comment]
But you are not combining rows here. What exact problem are you getting?

**Comment by muhammad-Ali-Nawaz** [comment]
yes on chrome its working fine but when we download csv from Mozilla firefox then the downloaded file will have all the data in a single row
